data/lua/core.lua

```diff
@@ -1,5 +1,7 @@
 -- Note: This file is loaded automatically by the engine.
 
+local helper = wesnoth.require("lua/helper.lua")
```

Comment:
I think this isn't a good idea because all other Lua code, including helper.lua, is allowed to assume that core.lua has already been loaded. That means that core.lua can't depend on helper.lua.

Instead, code that uses helper.lua here should be rewritten to not need it.

Reply:
@jyrkive I thought about it, too. The reason to include this line was this: https://github.com/wesnoth/wesnoth/blob/master/data/lua/core.lua#L96
It accesses helper anyway, but it assumes it's declared as a global variable already. It doesn't expect it to be initialised at core.lua init time though, to my commit might be a regression here.

Do you think we should re-write core.lua not to depend on helper.lua at all, or can we access it in a function (presumably later stage when all lua files have already being init) ?

Reply:
It shouldn't depend on helper.lua at all.

In this case, `helper.child_range()` is an alias to `wml.child_range()`, which is defined in _\*drum roll\*_ core.lua.

Reply:
lol, indeed. Fixed to use `wml.child_range`